### PR TITLE
Pushdown some unary operators.

### PR DESCRIPTION
**Pull request by Tony-Romanov**

### CODE REVIEW

File: ydb/library/yql/providers/common/pushdown/collection.cpp [hunk 1]
```diff
@@ -355,11 +351,11 @@ bool CheckExpressionNodeForPushdown(const TExprBase& node, const TExprNode* lamb
         return true;
     } else if (settings.IsEnabled(TSettings::EFeatureFlag::ParameterExpression) && node.Maybe<TCoParameter>()) {
         return true;
-    } else if (IsSupportedArithmeticalExpression(node, settings)) {
-        TCoBinaryArithmetic op = node.Cast<TCoBinaryArithmetic>();
-        return CheckExpressionNodeForPushdown(op.Left(), lambdaArg, settings) && CheckExpressionNodeForPushdown(op.Right(), lambdaArg, settings);
+    } else if (const auto op = node.Maybe<TCoUnaryArithmetic>(); op && settings.IsEnabled(TSettings::EFeatureFlag::ArithmeticalExpressions)) {
```
Comment (UgnineSirdis) [inline ydb/library/yql/providers/common/pushdown/collection.cpp:354]: This function is also called [here](https://github.com/ydb-platform/ydb/blob/main/ydb/library/yql/providers/generic/provider/yql_generic_physical_opt.cpp#L141C18-L141C18)
If you don't want to support a new expression [here](https://github.com/ydb-platform/ydb/blob/main/ydb/library/yql/providers/generic/provider/yql_generic_predicate_pushdown.h#L12) and futher throught the stack of generic provider, it's better to disable it with another flag in settings for generic provider and enable in OLAP

**Review by UgnineSirdis — CHANGES_REQUESTED** [review]

File: ydb/core/kqp/host/kqp_type_ann.cpp [hunk 2]
```diff
@@ -864,6 +864,21 @@ bool ValidateOlapFilterConditions(const TExprNode* node, const TStructExprType*
...
         return res;
+    } else if (TKqpOlapFilterUnaryOp::Match(node)) {
+        if (!EnsureArgsCount(*node, 2, ctx)) {
+            return false;
+        }
+        const auto op = node->Child(TKqpOlapFilterUnaryOp::idx_Operator);
+        if (!EnsureAtom(*op, ctx)) {
+            return false;
+        }
+        if (!op->IsAtom({"minus","abs","not","size","exists","empty"})) {
```
[the remark this answers is not in the document]
Comment (Tony-Romanov) [reply ydb/core/kqp/host/kqp_type_ann.cpp:875]: It's common replacement for these callables. I will not use them i future.

File: ydb/library/yql/providers/common/pushdown/collection.cpp (hunk 1 above)
Comment (Tony-Romanov) [reply ydb/library/yql/providers/common/pushdown/collection.cpp:354]: And where is tests for generic provider?

Comment (UgnineSirdis) [reply ydb/library/yql/providers/common/pushdown/collection.cpp]: `ydb/core/kqp/ut/federated_query/generic`
`ydb/core/kqp/ut/federated_query/generic_ut`

**Review by UgnineSirdis — APPROVED** [review]

**Pull request merged**